using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FoundationaLLM.Common.Exceptions
{
    /// <summary>
    /// Represents an error generated by a resource manager.
    /// </summary>
    public class ResourceProviderException : Exception
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ResourceProviderException"/> class with a default message.
        /// </summary>
        public ResourceProviderException()
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="ResourceProviderException"/> class with its message set to <paramref name="message"/>.
        /// </summary>
        /// <param name="message">A string that describes the error.</param>
        public ResourceProviderException(string? message) : base(message)
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="ResourceProviderException"/> class with its message set to <paramref name="message"/>.
        /// </summary>
        /// <param name="message">A string that describes the error.</param>
        /// <param name="innerException">The exception that is the cause of the current exception.</param>
        public ResourceProviderException(string? message, Exception? innerException) : base(message, innerException)
        {
        }
    }
}
